deployment/terraform/agent.go

```diff
@@ -58,7 +65,29 @@ func (t *Terraform) configureAndRunAgents(extAgent *ssh.ExtAgent) error {
 		commands = append([]string{"wget -O tmp.tar.gz " + t.config.LoadTestDownloadURL}, commands...)
 	}
 
-	for _, val := range t.output.Agents {
+	// If UsersFilePath is present, split the user credentials among all the agents,
+	// so that the logged in users don't clash
+	splitFiles := make([][]string, 0, len(t.output.Agents))
+	if t.config.UsersFilePath != "" {
```

Comment:
I wish we didn't have to resort to this. Ideally we upload the same file to all agents and make them aware of their own index (should already be doable with some regex) so that they can simply seek to the needed offset $\frac{nUsers}{nAgents} \cdot agentIdx$.

Reply:
Yeah, that's a good point. I'm going to merge this one as is, though, as it's already tested, and tackle your suggestion in https://mattermost.atlassian.net/browse/MM-54436